nixos/modules/services/misc/podgrab.nix

```diff
@@ -22,28 +24,59 @@ in
       example = 4242;
       description = lib.mdDoc "The port on which Podgrab will listen for incoming HTTP traffic.";
     };
+
+    dataDirectory = mkOption {
+      type = types.path;
+      default = "${stateDir}/data";
+      example = "/mnt/podcasts";
+      description = "Directory to store downloads.";
+    };
+
+    user = mkOption {
+      type = types.str;
+      default = "podgrab";
+      description = "User under which Podgrab runs, and which owns the download directory.";
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "podgrab";
+      description = "Group under which Podgrab runs, and which owns the download directory.";
+    };
   };
 
   config = lib.mkIf cfg.enable {
+    systemd.tmpfiles.settings."10-pyload" = {
+      ${cfg.dataDirectory}.d = { inherit (cfg) user group; };
+    };
+
     systemd.services.podgrab = {
       description = "Podgrab podcast manager";
       wantedBy = [ "multi-user.target" ];
       environment = {
-        CONFIG = "/var/lib/podgrab/config";
-        DATA = "/var/lib/podgrab/data";
+        CONFIG = "${stateDir}/config";
+        DATA = cfg.dataDirectory;
         GIN_MODE = "release";
         PORT = toString cfg.port;
       };
       serviceConfig = {
-        DynamicUser = true;
+        User = cfg.user;
```

Comment:
I think podgrab would be a great candidate for using systemd's `DyanamicUser` setting. Since you are already using StateDirectory, systemd will manage the permissions for you instead of needing a static user & group.

Reply:
Hmm, on second thought since `dataDirectory` may be moved outside of StateDirectory it is probably necessary to use a static user.

Unfortunately, I haven't found an easy way to move or bind mount StateDirectory to a different location.

Reply:
> I think podgrab would be a great candidate for using systemd's DyanamicUser setting. Since you are already using StateDirectory, systemd will manage the permissions for you instead of needing a static user & group.

I used to agree, that's why I initially wrote the module with `DynamicUser = true` :-)

> Hmm, on second thought since dataDirectory may be moved outside of StateDirectory it is probably necessary to use a static user.

Yes, precisely. As mentionned in the PR description, my goal is to use Podgrab in conjunction with Audiobookshelf, so being able to control the rights over the data directory becomes necessary.

All in all, I don't think it's a big loss to go from `DynamicUser` to a NixOS declarative `user`/`group`.